### ISSUE TITLE
Feature request: multi-threaded component rotation

### ISSUE THREAD

**Issue by andrasfuchs** [miho/freerouting#12]
It would be nice to have an option to select the components on the board, which orientation doesn't really matter for the author. 

This way the autorouter could iterate trough all the rotational combinations and check which arrangement is the best for routing. It's particularly useful for ICs with many pins.

If we allowed 0, 90, 180, 270 rotations, then if the number of components ready for rotation is 1, 2 or 3, the initial number of boards would become 4, 16, 64, because we would need to generate all board with all rotational combinations. Although this sounds a lot, it would mean that we could start the autorouters in parallel for each board, so we could finally take advantage of the multi-core processors, and it would result in only one board in the end, so it would not complicate the process for the user.

To implement this, we would need to complete the issues #8 first though, and we would probably need a board evaluation function as well, to make the router able to decide which arrangement and routing is better then the other.

**Comment by miho** [comment]
archived
